hack/arktos-worker-up.sh

```diff
@@ -47,29 +110,141 @@ if [[ ! -s "${KUBELET_CLIENTCA}" ]]; then
+
+    API_TENANT_SERVERS=(${API_TENANT_SERVER//,/ })
+    serverCount=${#API_TENANT_SERVERS[@]}
+    for (( pos=0; pos<${serverCount}; pos++ ));
+    do
+      # generate kubeconfig fpr tenant partitions
+      write_client_kubeconfig "${CONTROLPLANE_SUDO}" "${CERT_DIR}" "" "${API_TENANT_SERVERS[${pos}]}" "${API_PORT}" ${kubeconfig_filename} "" "http"
+      ${CONTROLPLANE_SUDO} mv "${CERT_DIR}/${kubeconfig_filename}.kubeconfig" "${CERT_DIR}/${kubeconfig_filename}${pos}.kubeconfig"
+      ${CONTROLPLANE_SUDO} chown "$(whoami)" "${CERT_DIR}/${kubeconfig_filename}${pos}.kubeconfig"
+
+      KUBELET_TENANT_SERVER_KUBECONFIG_FLAG="${KUBELET_TENANT_SERVER_KUBECONFIG_FLAG}${CERT_DIR}/${kubeconfig_filename}${pos}.kubeconfig,"
+      PROXY_TP_KUBECONFIGS="${PROXY_TP_KUBECONFIGS}${CERT_DIR}/${kubeconfig_filename}${pos}.kubeconfig,"
+    done
+    KUBELET_TENANT_SERVER_KUBECONFIG_FLAG=${KUBELET_TENANT_SERVER_KUBECONFIG_FLAG::-1}
+    PROXY_TP_KUBECONFIGS=${PROXY_TP_KUBECONFIGS::-1}
+  fi
+
+  sudo ./_output/local/bin/linux/amd64/hyperkube kubelet \
+  --v="${LOG_LEVEL}" \
+  --container-runtime=remote \
+  --hostname-override=${HOSTNAME_OVERRIDE} \
+  --address='0.0.0.0' \
+  --kubeconfig="${CERT_DIR}/kubelet.kubeconfig" \
+  ${KUBELET_TENANT_SERVER_KUBECONFIG_FLAG} \
+  --authorization-mode=Webhook \
+  --authentication-token-webhook \
+  --client-ca-file=${KUBELET_CLIENTCA} \
+  --feature-gates=${FEATURE_GATES} \
+  --cpu-cfs-quota=true \
+  --enable-controller-attach-detach=true \
+  --cgroups-per-qos=true \
+  --cgroup-driver= --cgroup-root= \
+  --cluster-dns=${CLUSTER_DNS} \
+  --cluster-domain=cluster.local \
+  --container-runtime-endpoint="containerRuntime,container,/run/containerd/containerd.sock;vmRuntime,vm,/run/virtlet.sock" \
+  --runtime-request-timeout=2m \
+  --port=10250 \
+  --resolv-conf=/run/systemd/resolve/resolv.conf \
+  > /tmp/kubelet.worker.log 2>&1 &
+
+fi
 
 echo "kubelet has been started. please check /tmp/kubelet.worker.log for its running log."
 
+if [ "${CNIPLUGIN}" == "flannel" ]; then
+    echo "DBG: Installing Flannel cni plugin..."
+    sleep 30  #need sometime for KCM to be fully functioning
+    if [ "${IS_SCALE_OUT}" == "true" ]; then
+       install_flannel "${RESOURCE_PARTITION_POD_CIDR}" "${API_HOST}"
+    else
+       install_flannel "${KUBE_CONTROLLER_MANAGER_CLUSTER_CIDR}" "${API_HOST}"
+    fi
+fi
+
+# Start proxy
+cat <<EOF > /tmp/kube-proxy.yaml
+apiVersion: kubeproxy.config.k8s.io/v1alpha1
+kind: KubeProxyConfiguration
+clientConnection:
+  kubeconfig: ${CERT_DIR}/kube-proxy.kubeconfig
+tenantPartitionKubeConfig: ${PROXY_TP_KUBECONFIGS}
+hostnameOverride: ${HOSTNAME_OVERRIDE}
+mode: ${KUBE_PROXY_MODE}
+EOF
+
+if [[ -n ${FEATURE_GATES} ]]; then
+  echo "featureGates:"
+  # Convert from foo=true,bar=false to
+  #   foo: true
+  #   bar: false
+  for gate in $(echo "${FEATURE_GATES}" | tr ',' ' '); do
+    echo "${gate}" | sed -e 's/\(.*\)=\(.*\)/  \1: \2/'
+  done
+fi >>/tmp/kube-proxy.yaml
```

Comment:
didn't see where to remove this /tmp/kube-proxy.yaml. this suppose to be deleted once finished usage.

Reply:
> didn't see where to remove this /tmp/kube-proxy.yaml. this suppose to be deleted once finished usage.

This file in in tmp folder, no need to worry about removing. Keep it there for debugging purpose.

Reply:
For cleanup purpose, I suggest we remove all files dynamic generated on dev machine to ensure next start will not be impacted by existing files.

Reply:
line 255 "cat <<EOF > /tmp/kube-proxy.yaml" is similar to clean old contents and fill with new contents.